### ISSUE TITLE
ParameterFilters in SSM describe_parameters does not support option Contains

### ISSUE THREAD

**Issue by JordanReiter** [spulec/moto#3159]
According to the current documentation for boto3, `Contains` is a valid filter if the value for Key is `Name`:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ssm.html#SSM.Client.describe_parameters

> For all filters used with DescribeParameters , valid options include `Equals` and `BeginsWith` . The `Name` filter additionally supports the `Contains` option.

Currently, moto does not support the `Contains` option for `describe_parameters`. The code in question does not check if the `Key` is `Name` (and therefore if you are able to use the `Contains` value).

**To Reproduce Issue**
```
with moto.mock_ssm():
    ssm_client = boto3.client('ssm')
    ssm_client.describe_parameters(ParameterFilters=[{'Key': "Name", 'Option': "Contains", 'Values': ["some_value"]}])
```

**Expected behavior**

Returns parameters where some part of the parameter name matches one of the values provided.

**Actual behavior**

Throws an exception:

```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/path/to/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/path/to/botocore/client.py", line 635, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidFilterOption: An error occurred (InvalidFilterOption) when calling the DescribeParameters operation: The following filter option is not valid: Contains. Valid options include: [BeginsWith, Equals].

```

**Behavior without mocking**
```
>>> ssm_client = boto3.client('ssm')
>>> ssm_client.describe_parameters(ParameterFilters=[{'Key': "Name", 'Option': "Contains", 'Values': ["some_value"]}])
{'Parameters': [], 'ResponseMetadata': {'RequestId': '...', 'HTTPStatusCode': 200, 'HTTPHeaders': {'server': 'Server', 'date': 'Mon, 20 Jul 2020 22:26:31 GMT', 'content-type': 'application/x-amz-json-1.1', 'content-length': '17', 'connection': 'keep-alive', 'x-amzn-requestid': '...'}, 'RetryAttempts': 0}}
```
**Version used**
'1.3.15.dev' (boto3==1.14.24, botocore==1.17.24)

**Comment by JordanReiter** [comment]
Please let me know if it would be helpful for me to offer a PR to implement this.

**Comment by bblommers** [comment]
Thanks for raising this @JordanReiter - will mark it as an enhancement.

PR's are always welcome! There are some guidelines in how to get started in our [Contributing.MD](https://github.com/spulec/moto/blob/master/CONTRIBUTING.md).

**Comment by JordanReiter** [comment]
Hi, I have written a fix for this but because I am working on this via my employer it has to go through review there before I can create a public PR. But just want you to know that I do have the changes including tests.

**Comment by mkielar** [comment]
I've just hit this today after naively hopping on `moto` (as SNS says `all endpoints done`) and half day of carefully crafting my mocks and unit tests... 

Anyway, two questions:
1. Does this MR also handle paginators?
2. Any chance this will this be released anytime soon? Last release is almost a year old now...

**Comment by bblommers** [comment]
Hi @mkielar , this does not handle paginators yet.
This functionality is already available via the dev-releases, as of moto >= 1.3.15.dev1007

Will close this, as the immediate functionality is now available. Feel free to open a new issue for any improvements you'd like to see.
